vendors/espressif/boards/ports/ota_pal_for_aws/ota_pal.c

```diff
@@ -21,10 +21,21 @@
 #include <stdio.h>
 #include <stdint.h>
 #include <string.h>
-#include "aws_iot_ota_agent.h"
-#include "aws_iot_ota_pal.h"
-#include "aws_iot_ota_interface.h"
-#include "aws_ota_agent_config.h"
+#include "ota.h"
+#include "ota_pal.h"
+#include "ota_interface_private.h"
+
+/* OTA_DO_NOT_USE_CUSTOM_CONFIG allows building the OTA library
+ * without a custom config. If a custom config is provided, the
+ * OTA_DO_NOT_USE_CUSTOM_CONFIG macro should not be defined. */
+#ifndef OTA_DO_NOT_USE_CUSTOM_CONFIG
+    #include "ota_config.h"
+#endif
+
+/* Include config defaults header to get default values of configs not defined
+ * in ota_config.h file. */
+#include "ota_config_defaults.h"
```

Comment:
Is this needed? I think this is already done through the ota.h header file. If it's not, can you consider removing this?

Reply:
Yes, this is a part of `ota_private.h` and not `ota.h` Hence it needs to be included here as well.

Reply:
ota.h includes ota_private.h at the top of it so it should still be okay.